File: app/src/main/java/blue/koenig/kingsfinances/model/calculation/FinanceStatisticsCalculatorService.java
package blue.koenig.kingsfinances.model.calculation;

import android.content.Context;

import com.koenig.commonModel.Byteable;

import java.nio.ByteBuffer;
import java.util.ArrayList;
import java.util.List;

import blue.koenig.kingsfamilylibrary.model.FamilyConfig;

/**
 * Created by Thomas on 09.01.2018.
 */

public class FinanceStatisticsCalculatorService implements StatisticsCalculatorService {

    private final Context context;
    private final String key;

    public FinanceStatisticsCalculatorService(Context context, String key) {
        this.context = context;
        this.key = key;
    }

    @Override
    public List<StatisticEntry> getSavedSortedStatistics() {
        ByteBuffer buffer = FamilyConfig.getBytesFromConfig(context, key);
        if (buffer == null) return new ArrayList<>();

        int size = buffer.getInt();
        List<StatisticEntry> debts = new ArrayList<>(size);
        for (int i = 0; i < size; i++) {
            debts.add(new StatisticEntry(buffer));
        }

        return debts;
    }

    @Override
    public void saveStatistics(List<StatisticEntry> statisticEntryList) {
        ByteBuffer buffer = ByteBuffer.allocate(Byteable.Companion.getBigListLength(statisticEntryList));
        Byteable.Companion.writeBigList(statisticEntryList, buffer);
        FamilyConfig.saveBytes(context, buffer.array(), key);
    }
}

File: app/src/main/java/blue/koenig/kingsfinances/model/calculation/FinanceStatisticsCalculatorService.kt
package blue.koenig.kingsfinances.model.calculation

import blue.koenig.kingsfinances.model.FinanceConfig
import com.koenig.commonModel.Byteable
import java.nio.ByteBuffer
import java.util.*

/**
 * Created by Thomas on 09.01.2018.
 */

class FinanceStatisticsCalculatorService(private val config: FinanceConfig, private val key: String) : StatisticsCalculatorService {

    override fun getSavedSortedStatistics(): List<StatisticEntryDeprecated> {
        val buffer = config.loadBuffer(key) ?: return ArrayList()

        val size = buffer.int
        val debts = ArrayList<StatisticEntryDeprecated>(size)
        for (i in 0 until size) {
            debts.add(StatisticEntryDeprecated(buffer))
        }

        return debts
    }

    override fun saveStatistics(statisticEntryList: List<StatisticEntryDeprecated>) {
        val buffer = ByteBuffer.allocate(Byteable.getBigListLength(statisticEntryList))
        Byteable.writeBigList(statisticEntryList, buffer)
        config.saveBuffer(buffer, key)
    }
}
